Revert "mac80211: replace legacy minstrel with minstrel_ht..."

thuehn opened the pull request:
This reverts commit 3a12c6679eb184ca2a9a2b1266cc2d41f3f00bad.

Initially I've tested this commit on my desktop-setup and it showed no errors with
AP-to-2x-legacy-STAs while different traffic loads.
Hence, I reported back to Felix that it looks good from my side for up-streaming it
to OpenWrt trunk.
But my current sysupgrade with this Minstrel patches included, 
to 25 PcEngines APU Boards in our Freifunk network triggered
occasional rcu stalls on some routers.
From my last working trunk state to the latest commit (bad state), I have 
git bisected the rcu stall problem been introduced by this commit.
Several productive nodes occasionally face a rcu
stall, hence an unresponsive/inaccessible OpenWrt. A power cycle is needed to
bring them back (without additional auto reboot scripting).
Our troubleshooting is ongoing... so for the time being lets revert this patch
until it's properly working on different  productive WiFi network setups.

Signed-off-by: Thomas Huehn <thomas.huehn@hs-nordhausen.de>

thuehn committed:
Revert "mac80211: replace legacy minstrel with minstrel_ht, improve rate selection"

This reverts commit 3a12c6679eb184ca2a9a2b1266cc2d41f3f00bad.

This commit was initially tested on my desktop setup and showed no errors in
AP-to-legacy-STA setups. Hence I reported Felix that it looks good from my side
for upstreaming it to OpenWrt trunk.
But my roll-out of this Minstrel patch to 25 PcEngines APU Boards in our
Freifunk network triggered occasional RCU stalls on those routers.
From my last working trunk to the latest commit, I have git bisected down
the culprit to this commit. Several productive nodes occasionally face a RCU
stall, hence an unresponsive/inacessible OpenWrt. A power cycle is needed to
bring them back without additional scripting.
Our troubleshooting is ongoing... so for the time lets revert this patch
until its working properly working in productive WiFi network setups.

Signed-off-by: Thomas Huehn <thomas.huehn@hs-nordhausen.de>

kofec commented:
Maybe this commit will solve your issue:
https://github.com/openwrt/openwrt/commit/faeaf5a01025edc13b83c90a724d8bff18ab2279

thuehn commented:
Hi all,

We have fixed the bug , tested the merged Minstrel patch set in different mesh & access loads.
Fixed in https://github.com/openwrt/openwrt/commit/faeaf5a01025edc13b83c90a724d8bff18ab2279.
Plz close this PR.

Greetings Thomas
